### PR TITLE
Enhanced bug report in github muC repo and in muC itself

### PR DESCRIPTION
**Pull request by pskowronek**
- For muC github repo added bug/issue & feature templates (fields based on RSyntaxTextArea repo)
- For muCommander - Report a bug now is able to pre-fill the following fields: muC version, java version, OS version, logs.

It's been tested on this repo:
https://github.com/pskowronek/parametrized_issue_template and a bug finally created looks like this: https://github.com/pskowronek/parametrized_issue_template/issues/2

Bug report page initiated from within muC looks like this then:
parametrized_issue_template - ok

### CODE REVIEW

**Review by ahadas — COMMENTED** [review]
this looks really nice @pskowronek!
it can save us from the common ping-pong for getting more information
the 'description', 'steps to reproduce', 'expected behavior' and 'actual behavior' are really mandatory but I think we can be a bit more tolerant about the others to ease filing issues - how about saying for the version that we fix only issues on the latest stable release and to ask users to specify a version only if they use a nightly build? I'd also make the Java version field non-mandatory as it's usually bundled now but to say "if non-bundled Java is used, please specify its version". and we can also make the OS field non-mandatory but that's less important because people should easily tell what operating system they use. wdyt?

**Comment by pskowronek** [comment]
> how about saying for the version that we fix only issues on the latest stable release and to ask users to specify a version only if they use a nightly build? I'd also make the Java version field non-mandatory as it's usually bundled now but to say "if non-bundled Java is used, please specify its version". and we can also make the OS field non-mandatory but that's less important because people should easily tell what operating system they use. wdyt?

From my observations we end up asking ppl for muC version or java version :) What I would change is the description - to something like this 'Please copy-paste the version shown in About'. We may also make it optional. WDYT?

File: mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java [hunk 1]
```diff
@@ -31,12 +37,55 @@
...
-        putValue(URL_PROPERTY_KEY, com.mucommander.RuntimeConstants.REPORT_BUG_URL);
+    @Override
+    public void performAction() {
+        Object url = com.mucommander.RuntimeConstants.REPORT_BUG_URL;
+        try {
+            // TODO remove - temporarily left for testing: uncomment to check on that test-repo
+            //url = "https://github.com/pskowronek/parametrized_issue_template/issues/new";
+            String ver = String.format(Activator.portable ? "%s (%s)" : "%s", RuntimeConstants.VERSION, Translator.get("portable"));
+            String java_ver = String.join(System.lineSeparator(),
+                    System.getProperty("java.version"),
```
Comment (pskowronek) [inline mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java]: And a common place to generate that info should be created to be also used by AboutDialog

Comment (ahadas) [reply mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java]: yeah, there's a room for refactoring there, maybe to use `JavaVersion#getRawSystemProperty`

**Comment by ahadas** [comment]
> From my observations we end up asking ppl for muC version or java version :) What I would change is the description - to something like this 'Please copy-paste the version shown in About'. We may also make it optional. WDYT?

yeah, makes sense - the easier it is to find and provide the information, the better

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      placeholder: Go to About of muCommander and copy-paste the version
+      value:
+    validations:
+      required: true
+  - type: textarea
+    id: java-version
+    attributes:
+      label: Java version
+      description: What version of Java muCommander is using?
+      placeholder: Go to About of muCommander and copy-paste the Java information
```
Comment (pskowronek) [inline .github/ISSUE_TEMPLATE/bug_report.yml:69]: I've already placed such information how to get the Version

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      value:
+    validations:
+      required: true
+  - type: textarea
+    id: os-version
+    attributes:
+      label: Operating System type and version
+      description: What type and version of Operating System you are running muCommander on?
+      placeholder: Go to About of muCommander and copy-paste the OS information
```
Comment (pskowronek) [inline .github/ISSUE_TEMPLATE/bug_report.yml:78]: and here, too

**Comment by pskowronek** [comment]
> > From my observations we end up asking ppl for muC version or java version :) What I would change is the description - to something like this 'Please copy-paste the version shown in About'. We may also make it optional. WDYT?
> 
> yeah, makes sense - the easier it is to find and provide the information, the better

See my comments to bug_report.yml - the information how to get versions is already there.

File: mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java (hunk 1 above)
Comment (pskowronek) [reply mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java]: Ok, I will take a look

Comment (pskowronek) [reply mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java]: I've taken a look - I'm not convinced to have any common place for those versions. I would leave it as is.

File: mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java [hunk 4]
```diff
@@ -1,45 +1,92 @@
...
+
+    public ReportBugAction(MainFrame mainFrame, Map<String,Object> properties) {
+        super(mainFrame, properties);
+    }
+
+    @Override
+    public void performAction() {
+        var url = com.mucommander.RuntimeConstants.REPORT_BUG_URL;
+        try {
+            var muCVersion = String.format(Activator.portable ? "%s (%s)" : "%s", RuntimeConstants.VERSION, Translator.get("portable"));
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/ui/action/impl/ReportBugAction.java:57]: why a translated form of 'portable'?

File: mucommander-core/src/main/java/com/mucommander/ui/dialog/about/AboutDialog.java
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/ui/dialog/about/AboutDialog.java:1]: can you please point out the changes in this file?

**Review by ahadas — COMMENTED** [review]
it seems like the entire AboutDialog file has changed - it would be great if you can help me identifying the changes that were done there, other than that lgtm

**Comment by pskowronek** [comment]
> it seems like the entire AboutDialog file has changed - it would be great if you can help me identifying the changes that were done there, other than that lgtm

See "this - REMOVED LINE" comment - this is the only change, the rest is CRLF->LF conversion plus tabs->spaces (the very first commit with cleaning only)

**Comment by ahadas** [comment]
> > it seems like the entire AboutDialog file has changed - it would be great if you can help me identifying the changes that were done there, other than that lgtm
> 
> See "this - REMOVED LINE" comment - this is the only change, the rest is CRLF->LF conversion plus tabs->spaces (the very first commit with cleaning only)

is it within the code or where can I find it?

**Comment by ahadas** [comment]
@pskowronek I tried to isolated the changes by applying the formatter and change CRLF to CL in AboutDialog but now the rebase fails on conflicts..

**Comment by ahadas** [comment]
@pskowronek oh, I see we updated the branch at the same time.. I reverted the changes to AboutDialog - could you post them in a separate PR?

**Review by ahadas — APPROVED** [review]
I reverted the changes in AboutDialog that were obscured by code refactoring - it doesn't seem to break anything so if the changes in AboutDialog are still relevant, please post them separately

**Pull request merged**